libs/cart/src/drivers/interfaces/cart-coupon-service.interface.ts

```diff
@@ -0,0 +1,34 @@
+import { InjectionToken } from '@angular/core';
+import { Observable } from 'rxjs';
+
+import { DaffCartCoupon } from '../../models/cart-coupon';
+import { DaffCart } from '../../models/cart';
+
+/**
+ * The interface responsible for applying a coupon to a cart.
+ */
+export interface DaffCartCouponServiceInterface<T extends DaffCart = DaffCart, V extends DaffCartCoupon = DaffCartCoupon> {
+	/**
+	 * Apply a coupon to the cart and return a partial of the cart.
+	 */
+	apply(cartId: T['id'], coupon: V): Observable<Partial<T>>;
+
+	/**
+	 * List coupon codes applied to a cart.
+	 */
+	list(cartId: T['id']): Observable<Partial<T>>;
```

Comment:
I think the return type should be `Observable<V[]>`.

Reply:
yea, mb. Changed